src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/InvokerUtil.cs

```diff
@@ -150,8 +166,81 @@ internal static InvokeDelegate GenerateInvokeDelegate(MethodInfo method, out int
+            {
+                inputParameterCount = 0;
+                outputParameterCount = 0;
+                ParameterInfo[] parameters = method.GetParameters();
+                bool returnsValue = method.ReturnType != typeof(void);
+
+                var targetParam = Expression.Parameter(typeof(object), "target");
+                var inputsParam = Expression.Parameter(typeof(object[]), "inputs");
+                var outputsParam = Expression.Parameter(typeof(object[]), "outputs");
+
+                List<ParameterExpression> variables = new();
+                var result = Expression.Variable(typeof(object), "result");
+                variables.Add(result);
+
+                List<ParameterExpression> outputVariables = new();
+                List<ParameterExpression> invocationParameters = new();
+                List<Expression> expressions = new();
+
+                for (int i = 0; i < parameters.Length; i++)
+                {
+                    Type variableType = parameters[i].ParameterType.IsByRef
+                        ? parameters[i].ParameterType.GetElementType()
+                        : parameters[i].ParameterType;
+                    ParameterExpression variable = Expression.Variable(variableType, $"p{i}");
+
+                    if (ServiceReflector.FlowsIn(parameters[i]))
+                    {
+                        expressions.Add(Expression.Assign(variable, Expression.Convert(Expression.ArrayIndex(inputsParam, Expression.Constant(inputParameterCount)), variableType)));
+                        inputParameterCount++;
+                    }
+
+                    if (ServiceReflector.FlowsOut(parameters[i]))
+                    {
+                        outputParameterCount++;
+                        outputVariables.Add(variable);
+                    }
+
+                    variables.Add(variable);
+                    invocationParameters.Add(variable);
+                }
+
+                var castTargetParam = Expression.Convert(targetParam, method.DeclaringType);
+
+                expressions.Add(returnsValue
+                    ? Expression.Block(Expression.Assign(result, Expression.Convert(Expression.Call(castTargetParam, method, invocationParameters), typeof(object))))
+                    : Expression.Block(Expression.Call(castTargetParam, method, invocationParameters),
+                            Expression.Assign(result, Expression.Constant(null, typeof(object)))));
+
+                int j = 0;
+                foreach (var outputVariable in outputVariables)
+                {
+                    expressions.Add(Expression.Assign(
+                        Expression.ArrayAccess(outputsParam, Expression.Constant(j)),
+                        Expression.Convert(outputVariable, typeof(object))));
+                    j++;
+                }
+
+                expressions.Add(Expression.Block(
+                    Expression.Empty(),
```

Comment:
I think you can change this to just
```c#
expressions.Add(result);
```
As the finalBlock will return the value of the last expression. This is nesting a block inside a block and I don't think that's needed.

Reply:
done